### PR TITLE
Fix typo and find command bug and improve code quality

### PR DESCRIPTION
**Pull request by dhlee03**
## Changes:

- Each people seems to include field name inside information used for find command, which results in bug. Field name is removed.
- Improve overall codequality
- Fix typo
- Update app version
- Fix bug in link command: previous regex accepts zero as valid index which resulted in index error, causing app to crash. Now it prevents zero input.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/LinkCommand.java
Comment (julia-cp) [inline src/main/java/seedu/address/logic/commands/LinkCommand.java:1]: check index==0 for validateIndex

File: src/main/java/seedu/address/logic/commands/LinkCommand.java [hunk 1]
```diff
@@ -21,58 +24,114 @@ public class LinkCommand extends Command {
...
+     *
+     * @param indexes the indexes of the selected people
+     * @param persons the list of all people
+     * @throws CommandException if the indexes are invalid
+     */
+    private void validateIndexes(List<Index> indexes, List<Person> persons) throws CommandException {
+        for (Index index : indexes) {
+            boolean isNegative = index.getZeroBased() < 0;
+            boolean isOver = index.getZeroBased() >= persons.size();
+            if (isNegative || isOver) {
```
Comment (julia-cp) [inline src/main/java/seedu/address/logic/commands/LinkCommand.java:73]: check index==0

Comment (dhlee03) [reply src/main/java/seedu/address/logic/commands/LinkCommand.java:73]: Index is zero-based, so zero is acceptable value. However, there was another problem with parser regex so I fixed that part.

File: src/main/java/seedu/address/logic/parser/LinkCommandParser.java [hunk 2]
```diff
@@ -16,17 +19,19 @@ public class LinkCommandParser implements Parser<LinkCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public LinkCommand parse(String args) throws ParseException {
-        Boolean isEmpty = args.trim().isEmpty();
-        Boolean isInvalid = !args.trim().matches("\\d+(\\s+\\d+)*");
+        String trimmed = args.trim();
+        Boolean isEmpty = trimmed.isEmpty();
+        String validationRegex = "[1-9][0-9]*(\\s+[1-9][0-9]*)*";
```
Comment (bachletuankhai) [inline src/main/java/seedu/address/logic/parser/LinkCommandParser.java:24]: Not too sure about this. What if the input is 02? It should be ok right?

**Review by bachletuankhai — COMMENTED** [review]
The regex is a bit weird. Should you just use `ParserUtil::parseIndex` for index?

File: src/main/java/seedu/address/logic/parser/LinkCommandParser.java (hunk 2 above)
Comment (dhlee03) [reply src/main/java/seedu/address/logic/parser/LinkCommandParser.java:24]: 02 is not a valid input in normal cases. we can specify that in our UG.

Comment (bachletuankhai) [reply src/main/java/seedu/address/logic/parser/LinkCommandParser.java:24]: It is a valid input as index in commands like `delete` or `edit`. There is a parser class method for parsing index, I think you can use that.

**Review by bachletuankhai — APPROVED** [review]
LGTM

**Pull request merged**